geomstats/learning/pca.py

```diff
@@ -353,3 +355,186 @@ def _fit(self, X, base_point=None):
         self.singular_values_ = singular_values_[:n_components]
 
         return U, S, V
+
+
+class HyperbolicPlanePCA(_BasePCA):
+    """Exact Principal Geodesic Analysis in the hyperbolic plane.
+
+    The first principal component is computed by finding the direction
+    in a unit ball around the mean that maximizes the variance of the
+    projections on the induced geodesic. The projections are given by
+    closed form expressions in extrinsic coordinates. The second principal
+    component is the direction at the mean that is orthogonal to the first
+    principal component.
+
+    Parameters
+    ----------
+    space : Hyperbolic
+        Two-dimensional hyperbolic space.
+    n_vec : int
+        Number of vectors used to discretize the unit ball when finding
+        the direction of maximal variance.
+
+    Attributes
+    ----------
+    components_ : array-like, shape=[n_components, 2]
+        Principal axes, representing the directions of maximal variance in the
+        data. They are the initial velocities of the principal geodesics.
+    mean_ : array-like, shape=[2,]
+        Intrinsic mean of the data points.
+
+    References
+    ----------
+    .. [CSV2016] R. Chakraborty, D. Seo, and B. C. Vemuri,
+        "An efficient exact-pga algorithm for constant curvature manifolds."
+        Proceedings of the IEEE conference on computer vision and pattern
+        recognition. 2016.
+    """
+
+    def __init__(self, space, n_vec=100):
+        self.space = space
+        self.half_space = Hyperbolic(2, coords_type="half-space")
+        self.space_ext = Hyperbolic(2, coords_type="extrinsic")
+        self.n_vec = n_vec
+
+    def _project_on_geodesic_extrinsic(self, point_ext, mean_ext, vector_ext):
```

Comment:
Shouldn't we move this method to `Hyperboloid`? It looks like an interest method to have there.